Tests/web3swiftTests/localTests/EventTests.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  EventTests.swift
+//
+//
+//  Created by liugang zhang on 2023/8/24.
+//
+
+import XCTest
+import Web3Core
+
+@testable import web3swift
+
+class EventTests: XCTestCase {
+    func testEncodeTopc() throws {
+        let encoder = JSONEncoder()
+        let t1: [EventFilterParameters.Topic] = []
+        let t2: [EventFilterParameters.Topic] = [.string(nil)]
+        let t3: [EventFilterParameters.Topic] = [.strings([.string(nil), .string("1")])]
+        XCTAssertNoThrow(try encoder.encode(t1))
+        XCTAssertNoThrow(try encoder.encode(t2))
+        XCTAssertNoThrow(try encoder.encode(t3))
+
+        let t4: [EventFilterParameters.Topic] = [
+            .string("1"),
+            .strings([
+                .string("2"),
+                .string("3"),
+            ]
+        )]
+        let encoded = try encoder.encode(t4)
+        let json = try JSONSerialization.jsonObject(with: encoded)
+        XCTAssertEqual(json as? NSArray, ["1", ["2", "3"]])
+    }
+
+    func testEncodeLogs() throws {
+        let contract = try EthereumContract(TestEvent)
+        let logs = contract.events["UserOperationEvent"]?.encodeParameters(
+            [
+                "0x2c16c07e1c68d502e9c7ad05f0402b365671a0e6517cb807b2de4edd95657042",
+                "0x581074D2d9e50913eB37665b07CAFa9bFFdd1640",
+                "hello,world",
+                true,
+                "0x02c16c07e1c68d50",
+                nil
+            ]
+        )
+
+        XCTAssert(logs?.count == 7)
+    }
+
+    let TestEvent = """
```

Comment:
Please:
- move it to the top to follow the same style as in the other files;
- rename it to `testEvent`.

Reply:
Hmm, I'd even think about moving that into `testEncodeLogs` and adding a description that this ABi contains all `ABI.Element.Event.Input` types we want to test.

I'm thinking that it would be also good to test `func encodeTopic(input: ABI.Element.Event.Input, value: Any)` in isolation passing manually types and values and checking that the output is exactly what we expect.